tests/filecheck/transforms/scf_for_loop_range_folding.mlir

```diff
@@ -1,95 +1,56 @@
 // RUN: xdsl-opt -p scf-for-loop-range-folding --split-input-file %s | filecheck %s
 
-func.func @fold_one_loop(%arg0: memref<?xi32>, %arg1: index, %arg2: index) {
-  %c0 = arith.constant 0 : index
-  %c1 = arith.constant 1 : index
-  %c4 = arith.constant 4 : index
-  scf.for %i = %c0 to %arg1 step %c1 {
-    %0 = arith.addi %arg2, %i : index
-    %1 = arith.muli %0, %c4 : index
-    %2 = memref.load %arg0[%1] : memref<?xi32>
-    %3 = arith.muli %2, %2 : i32
-    memref.store %3, %arg0[%1] : memref<?xi32>
-  }
-  return
+// CHECK:       %new_lb = arith.addi %lb, %shift : index
+// CHECK-NEXT:  %new_ub = arith.addi %ub, %shift : index
+// CHECK-NEXT:  %new_lb_1 = arith.muli %new_lb, %mul_shift : index
+// CHECK-NEXT:  %new_ub_1 = arith.muli %new_ub, %mul_shift : index
+// CHECK-NEXT:  %new_step = arith.muli %step, %mul_shift : index
+// CHECK-NEXT:  scf.for %i = %new_lb_1 to %new_ub_1 step %new_step {
+// CHECK-NEXT:    "test.op"(%i) : (index) -> ()
+// CHECK-NEXT:  }
+
+%shift, %mul_shift, %lb, %ub, %step = "test.op"() : () -> (index, index, index, index, index)
+scf.for %i = %lb to %ub step %step {
+    %0 = arith.addi %shift, %i : index
+    %1 = arith.muli %0, %mul_shift : index
```

Comment:
So wrt my comment below, if we give names to these variables, their names will be visible outside of the for loop in the added arithmetic calculations

Reply:
Could you please check if that's what you meant?